tests/pipeline/nodes/test_base.py

```diff
@@ -0,0 +1,517 @@
+# Copyright 2022 AI Singapore
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from pathlib import Path
+from unittest import mock
+
+import pytest
+
+from peekingduck.pipeline.nodes.base import (
+    PEEKINGDUCK_WEIGHTS_SUBDIR,
+    ThresholdCheckerMixin,
+    WeightsDownloaderMixin,
+)
+from tests.conftest import PKD_DIR, do_nothing, not_raises
+
+
+@pytest.fixture
+def weights_model():
+    return WeightsModel()
+
+
+@pytest.fixture
+def threshold_model():
+    return ThresholdModel()
+
+
+class WeightsModel(WeightsDownloaderMixin):
+    def __init__(self):
+        self.config = {}
+        self.logger = mock.Mock()
+
+
+class ThresholdModel(ThresholdCheckerMixin):
+    def __init__(self):
+        self.config = {"a": 12.5, "b": 10, "c": "value", "d": [10, 12.5, 13]}
+
+
+class TestBase:
```

Comment:
Minor point for consideration - perhaps can split into 2 classes, one for WeightsDownloaderMixin and one for ThresholdCheckerMixin.

Reply:
Split to 2 test files